test/Microsoft.AspNetCore.Sockets.Common.Tests/Internal/Formatters/ServerSentEventsParserTests.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Text;
+using Microsoft.AspNetCore.Sockets.Internal.Formatters;
+using Xunit;
+using System.Threading.Tasks;
+using System.IO.Pipelines;
+
+namespace Microsoft.AspNetCore.Sockets.Common.Tests.Internal.Formatters
+{
+    public class ServerSentEventsParserTests
+    {
+        [Theory]
+        [InlineData("data: T\r\ndata: Hello, World\r\n\r\n", MessageType.Text, "Hello, World")]
+        [InlineData("data: T\r\ndata: Hello\r\ndata: , World\r\n\r\n", MessageType.Text, "Hello, World")]
+        public async Task ParseSSEMessage(string encodedMessage, MessageType messageType, string expectedMessage)
```

Comment:
Yeah, try to hit every branch in your parser with a test. See the tests for the existing formatters and parsers for some ideas of how to mess with the parser :)